lambda_function.py

```diff
@@ -0,0 +1,93 @@
+import json
+import os
+import logging
+from google.cloud import bigquery
+
+
+def lambda_handler(event):
+    """
+    Parses messages sent to liveclassAttendanceEventHandler lambda function..
+    Each message needs to have the following fields for row to be inserted:
+        - dateTime
+        - purpose
+            - type
+            - subType
+            - params
+                - platform
+                - id
+        - authType
+        - user
+            - values
+            - userDataValidated
+    """
+    messageBody = event["Records"]
+
+    for eachMessage in messageBody:
+        if "Sns" in eachMessage and "Message" in eachMessage["Sns"]:
+            # extracting the body of the message
+            message = json.loads(eachMessage["Sns"]["Message"])[0]
+
+            row = {}
+
+            # check if the key values exist in the message sent
+            # if all the fields don't exist,
+            #   the row isn't inserted and an error is displayed
+
+            if all(
+                (k in message for k in ("dateTime", "purpose", "authType", "user"))
+                and (k in message["purpose"] for k in ("type", "subType", "params"))
+                and (k in message["user"] for k in ("values", "userDataValidated"))
+                and (k in message["purpose"]["params"] for k in ("platform", "id"))
+            ):
+
+                # the key values are the column names in the BQ table.
+                row["timestamp"] = message["dateTime"]
+                row["purpose_type"] = message["purpose"]["type"]
+                row["purpose_sub_type"] = message["purpose"]["subType"]
+                row["platform"] = message["purpose"]["params"]["platform"]
+                row["platform_id"] = message["purpose"]["params"]["id"]
+                row["auth_type"] = message["authType"]
+                row["user_id"] = message["user"]["values"]
+                row["user_data_validated"] = message["user"]["userDataValidated"]
+
+                return insert_data(row)
+
+            else:
+                logging.error(
+                    "Encountered missing fields in message: {}".format(message)
+                )
+
+        else:
+            logging.error(
+                "Encountered missing fields in message: {}".format(eachMessage)
+            )
+
+
+def insert_data(row):
+    """
+    Function which inserts row into bigquery.
+    Project ID, Dataset ID, Table ID ae all stored as .env variables.
+    """
+
+    # load env variables
+    project_id = os.environ.get("BIGQUERY_PROJECT_ID")
```

Comment:
How are you getting environment variables on to Lambda?

Reply:
AWS Lambda has the functionality to add environment variables. There is a section in Configuration > Environment Variables.